### PR TITLE
Update threat model

### PR DESCRIPTION
**Pull request by ferozsalam**
- Incorporate feedback on transparent encryption with regards to network attackers
- Add recommended controls for Hubble
- Also some minor formatting tweaks to bring the document in line with our style guide

### CODE REVIEW

File: Documentation/security/threat-model.rst [hunk 1]
```diff
@@ -407,30 +407,11 @@ The threat matrix for such an attacker is as follows:
...
-|                  |    they could also spoof traffic by spoofing      |
-|                  |    pod identities.                                |
-|                  |                                                   |
-|                  | Denial of service could occur depending on the    |
-|                  | behavior of the attacker.                         |
+| Network data     | - Without transparent encryption, an attacker     |
+|                  |   could inspect traffic between workloads in both |
+|                  |   overlay and native routing modes.               |
+|                  | - Denial of service could occur depending on the  |
+|                  |   behavior of the attacker.                       |
```
Comment (zacharysarah) [inline Documentation/security/threat-model.rst:414]: Nice! ✨

File: Documentation/security/threat-model.rst [hunk 2]
```diff
@@ -725,21 +687,23 @@ Overall Recommendations
 To summarize the recommended controls to be used when configuring a
 production Kubernetes cluster with Cilium:
 
-1. Ensure that Kubernetes roles are scoped correctly to the requirements of your
+#. Ensure that Kubernetes roles are scoped correctly to the requirements of your
```
Comment (zacharysarah) [inline Documentation/security/threat-model.rst:690]: Good practice on numbering. Have you verified that your local build renders ordered steps correctly?

**Review by zacharysarah — APPROVED** [review]
LVGTM from a docs perspective. ✨ 

Screenshot 2023-04-05 at 3 03 38 PM

😻

**Review by joestringer — APPROVED** [review]

File: Documentation/security/threat-model.rst (hunk 2 above)
Comment (ferozsalam) [reply Documentation/security/threat-model.rst:690]: Yep, all looks good to me!

**Review by pchaigno — APPROVED** [review]
Thanks Feroz!

(Completely unrelated: your @isovalent.com email address isn't linked to your GitHub account.)

**Pull request merged**